dffml/source/source.py

```diff
@@ -29,12 +29,45 @@ def __init__(self, parent: "BaseSource") -> None:
     async def update(self, record: Record):
         """
         Updates a record for a source
+
+        Examples
+        --------
+        >>> async def main():
+        ...     async with ExampleSource(records=[]) as testSource:
+        ...         # Open, update, and close
+        ...         async with testSource() as sourceContext:
+        ...             example = Record("one")
+        ...             # ... Update one into our records ...
+        ...             await sourceContext.update(example)
+        ...             # Let's check out our records after calling `record` and `update`.
+        ...             ITER = sourceContext.records()
+        ...             async for ITEM in ITER:
+        ...                 print(ITEM)
+        >>>
+        >>> asyncio.run(main())
+        one
+        Predictions
+        Undetermined
         """
 
     @abc.abstractmethod
     async def records(self) -> AsyncIterator[Record]:
         """
         Returns a list of records retrieved from self.src
+
+        Examples
+        --------
+        >>> async def main():
+        ...     async with ExampleSource(records=[Record("example")]) as testSource:
+        ...         async with testSource() as sourceContext:
+        ...             ITER = sourceContext.records()
+        ...             async for ITEM in ITER:
```

Comment:
Looking good, let's stay away from uppercase local variables, and change up the names here a bit.

```suggestion
        ...             async for record in ctx.records():
```

Reply:
Also, let's not make the call to `ctx.records()` a separate line, that is likely to confuse people